framework/auth/core.py

```diff
@@ -1073,20 +1068,27 @@ def profile_url(self):
         return '/{}/'.format(self._id)
 
     @property
+    def contributed(self):
+        from website.project.model import Node
+        return Node.find(Q('contributors', 'eq', self._id))
+
+    @property
     def contributor_to(self):
-        return (
-            node for node in self.contributed
-            if not (
-                node.is_deleted
-                or node.is_bookmark_collection
-            )
+        from website.project.model import Node
+        return Node.find(
+            Q('contributors', 'eq', self._id) &
+            Q('is_deleted', 'ne', True) &
+            Q('is_bookmark_collection', 'ne', True)
```

Comment:
`is_collection` will also catch `is_bookmark_collection`, so that's probably the best way to go.